### PR TITLE
(DOCSP-19327): [Swift SDK] Draft Bundle a Realm page for bundling synced realms

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-19327

### Staged Changes (Requires MongoDB Corp SSO)

- [Bundle a Realm -> Swift SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-19327/sdk/swift/examples/bundle-a-realm/)

---

### Before merging:
- [x] Update `..versionadded` to reflect the release version
- [x] Update Podfile to point to release version
- [x] Add link from ``writeCopy`` and ``seedFilePath`` to generated SDK reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/ios/Examples/BundleRealms.swift [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import RealmSwift
+import Foundation
+
+class BundleRealms: XCTestCase {
+
+    func testCopyRealm() async throws {
+        // :code-block-start: copy-synced-realm-for-bundling
+        let app = App(id: YOUR_REALM_APP_ID)
+
+        // Log in the user whose Realm you want to copy for bundling
```
Comment (nathan-contino-mongo) [inline examples/ios/Examples/BundleRealms.swift:17]:
```suggestion
        // Log in the user whose realm you want to copy for bundling
```
maybe i should just give up the lowercase realm effort

File: examples/ios/Examples/BundleRealms.swift [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        // Create a configuration for the app user's realm
+        // This should use the same partition value as the bundled Realm
+        var config = user.configuration(partitionValue: "Partition You Want to Bundle")
+        config.objectTypes = [QsTask.self]
+
+        // Specify the path for the app user's realm
+        // We're using FileManager here for tested code examples,
+        // but you'll use NSBundle.main.path(forResource: ofType)
+        // if you bundle the realm with your app via Xcode.
+        guard let seedFileDir = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first else { return }
```
Comment (nathan-contino-mongo) [inline examples/ios/Examples/BundleRealms.swift:87]: I wonder if it makes more sense to make a bundled copy of this realm, like what I do for the Android SDK, and then open that bundled copy -- that way you don't need this comment/workaround (and can be sure you test real bundling code!)

File: examples/ios/Examples/BundleRealms.swift [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        if Realm.fileExists(for: config) {
+            try Realm.deleteFiles(for: config)
+            print("Successfully deleted existing realm at path: \(bundleRealmFilePath)")
+        } else {
+            print("No file currently exists at path")
+        }
+
+        // Write a copy of the realm you want to bundle at the path you specified
+        try realm.writeCopy(toFile: bundleRealmFilePath)
+        print("Successfully made a copy of the realm at path: \(bundleRealmFilePath)")
```
Comment (nathan-contino-mongo) [inline examples/ios/Examples/BundleRealms.swift:67]: Thought: should we check to make sure that this file exists afterwards in an assertion?

File: examples/ios/Podfile [hunk 4]
```diff
@@ -13,14 +13,14 @@ target 'QuickStartSwiftUI' do
...
-  pod 'RealmSwift', '=10.22.0'
+  pod 'RealmSwift', :git => 'https://github.com/realm/realm-swift.git', :branch => 'master'
 end
 
 target 'SwiftUIExamples' do
   use_frameworks!
 
   pod 'SwiftLint'
-  pod 'RealmSwift', '=10.22.0'
+  pod 'RealmSwift', :git => 'https://github.com/realm/realm-swift.git', :branch => 'master'
```
Comment (nathan-contino-mongo) [inline examples/ios/Podfile:23]: Boy do I wish there was a KMM/Java equivalent of this!

File: source/sdk/swift/examples/bundle-a-realm.txt [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. versionadded:: ?
+
+   Swift SDK version ? introduced the ability to bundle synchronized 
+   {+realm+}s. Before version ?, you could only bundle local {+realm+}s.
```
Comment (nathan-contino-mongo) [inline source/sdk/swift/examples/bundle-a-realm.txt:18]: Boy does this directive not work with body text! I've just used notes with the title "added in version <>" before to get a proper title + explanation before for version info.

Comment (nathan-contino-mongo) [reply source/sdk/swift/examples/bundle-a-realm.txt:18]: LOL it seems I did do exactly this in the android version of this page: https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-19327/sdk/android/examples/bundle-a-realm/

**Review by nathan-contino-mongo — APPROVED** [review]
LGTM with a couple of small nits, most notable of which is the "maybe we should bundle a realm with the example app" suggestion. Overall this makes a lot of sense to me and I like the proper testing in your example snippets!

File: examples/ios/Examples/BundleRealms.swift (hunk 1 above)
Comment (dacharyc) [reply examples/ios/Examples/BundleRealms.swift:17]: Hah, no, you're right, I'm really bad about consistency w/Realm vs. realm. I'll clean up the instances in this PR. I should really make a checklist for myself to run through when making PRs and add this to it...

File: examples/ios/Podfile (hunk 4 above)
Comment (dacharyc) [reply examples/ios/Podfile:23]: It's pretty great, but is annoying to switch it for drafting the docs while feature is merged but not released, and then switching again after feature is released. And occasionally the workspace gets into a weird state and I have to deintegerate the Pods and delete all the versions and then pod install again. Swift Package Manager is a bit nicer in that regard, and I have a ticket to switch it over at some point, but it never quite gets to the top of the priority list.

File: source/sdk/swift/examples/bundle-a-realm.txt (hunk 5 above)
Comment (dacharyc) [reply source/sdk/swift/examples/bundle-a-realm.txt:18]: Oof, yeah, the styling in that could be improved. I had forgotten. Will switch it.

File: examples/ios/Examples/BundleRealms.swift (hunk 2 above)
Comment (dacharyc) [reply examples/ios/Examples/BundleRealms.swift]: I have updated this to show the bundled version. Good call!

File: examples/ios/Podfile [hunk 6]
```diff
@@ -3,8 +3,8 @@ platform :ios, '15.0'
 target 'RealmExamples' do
   use_frameworks!
   pod 'SwiftLint'
-  pod 'Realm', '=10.22.0'
-  pod 'RealmSwift', '=10.22.0'
+  pod 'Realm', :git => 'https://github.com/realm/realm-swift.git', :branch => 'master'
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply examples/ios/Podfile:6]: Yeah, that has become our practice; we point to master to play with the code and write the tests, but wait for the release and then update the Podfile and the places in docs where we point to version-specific info before actually publishing. I've got a checklist in the "Before merging" section of the PR description above. Good call, though!

**Pull request merged**